sympy/core/mod.py

```diff
@@ -117,15 +117,17 @@ def doit(p, q):
                 net = Add(*non_mod_l) + Add(*[i.args[0] for i in mod_l])
                 return cls(net, q)
 
-        elif isinstance(p, Mul):
+        if isinstance(p, Mul):
```

Comment:
import `denom` from radsimp and then make this `if isinstance(p, Mul) and denom(p) == 1:` since only if a and b are coprime will `Mod(a/b,n)==Mod(a,n)*Mod(1/b,n)`.